lib/rbeapi/api/routemaps.rb

```diff
@@ -43,63 +43,439 @@ module Api
     # the basis of such criteria as route metrics, access control lists, next
     # hop addresses, and route tags.
     #
-    # rubocop:disable Metrics/MethodLength
-    #
     class Routemaps < Entity
+      ##
+      # get returns the routemap configuration
+      #
+      # @example
+      #   {
+      #     action: <string>,
+      #     seqno: <integer>,
+      #     match_rules: <array>,
+      #     set_rules: <array>,
+      #     continue: <integer>
+      #     description: <string>
+      #   }
+      #
+      # @param [String] name The routemap name to return a resource for from the
+      #   nodes configuration
+      #
+      # @return [nil, Hash<Symbol, Object>] Returns the routemap resource as a
+      #   Hash. If the specified name is not found in the nodes current
+      #   configuration a nil object is returned
       def get(name)
-        entries = config.scan(/^route-map\s#{name}\s.+$/)
+        entries = entries_regex(name)
+        return nil unless entries
+        parse_entries(entries)
+      end
 
-        entries.each_with_object([]) do |rm, arry|
+      ##
+      # getall returns a collection of routemap resource hashes from the nodes
+      # running configuration. The routemap resource collection hash is keyed
+      # by the unique routemap name.
+      #
+      # @example
+      #   [
+      #     <test:10>: {
+      #       action: <string>,
+      #       seqno: <integer>,
+      #       match_rules: <array>,
+      #       set_rules: <array>,
+      #       continue: <integer>
+      #       description: <string>
+      #     },
+      #     ...
+      #   ]
+      #
+      # @return [Hash<Symbol, Object>] returns a hash that represents the
+      #   entire routemap collection from the nodes running configuration.  If
+      #   there are no routemap names configured, this method will return an
+      #   empty hash.
+      def getall
+        routemaps = config.scan(/(?<=^route-map\s)[^\s]+/)
+        return nil unless routemaps
+        response = {}
+        routemaps.each do |name|
+          entries = entries_regex(name)
+          response[name] = parse_entries(entries)
+        end
+        response
+      end
+
+      ##
+      # entries regex scans the config for the specified route-name. The method
+      #   reduces duplication.
+      #
+      def entries_regex(name)
```

Comment:
Why isn't this code in parse_entries? The pattern you have is:
entries = entries_regex(name)
parse_entries(entries)